### PR TITLE
Pool: adds a validation and panic for when shares are equal to zero.

### PR DESCRIPTION
**Pull request by gangov**

### CODE REVIEW

**Comment by gangov** [comment]
if we do something like 

```Rust
let mut share_ratio = Decimal::zero();
let total_shares = utils::get_total_shares(&env);
if total_shares != 0i128 {
  share_ratio = Decimal::from_ratio(share_amount, total_shares);
} else {
  panic_with_err!(..)
}
```
then this makes the assignment of `Decimal::zero();` to `share_ratio` conflict with the compiler. 

Using two separate if statements seems too much for me, so that's why I go with this in the PR

File: contracts/pool/src/contract.rs [hunk 1]
```diff
@@ -415,12 +415,14 @@ impl LiquidityPoolTrait for LiquidityPool {
         let pool_balance_a = utils::get_pool_balance_a(&env);
         let pool_balance_b = utils::get_pool_balance_b(&env);
 
-        let mut share_ratio = Decimal::zero();
         let total_shares = utils::get_total_shares(&env);
-        if total_shares != 0i128 {
-            share_ratio = Decimal::from_ratio(share_amount, total_shares);
+
+        if total_shares == 0i128 {
+            panic_with_error!(env, ContractError::TotalSharesEqualZero);
```
Comment (ueco-jb) [inline contracts/pool/src/contract.rs:421]: 1) Stay consistent with the rest of the code base.
2) Notice, that the auditor reported a very important problem - if in this place total shares are equal to 0, it's a critical issue within the whole protocol. It should be exposed more.
```suggestion
        log!(
            &env,
            "Pool: WithdrawLiquidity: Critical error - Total shares are equal to zero before withdrawal!"
        );
        if total_shares == 0i128 {
            panic_with_error!(env, ContractError::TotalSharesEqualZero);
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -32,8 +32,10 @@ and this project adheres to
 ## Bug fixes
 
 - Pool stable: Fixes an error in the compute_swap function, where commission isn't deducted ([233])
+- Pool: Adds a validation for when shares can be zero ([#245])
```
Comment (ueco-jb) [inline CHANGELOG.md:35]:
```suggestion
- Pool: Adds a validation for when shares can be zero during withdrawal ([#245])
```

File: contracts/pool/src/storage.rs [hunk 3]
```diff
@@ -253,12 +253,14 @@ pub mod utils {
...
             if min_a > desired_a {
+                log!(&env, "Pool: GetDepositAmounts: Critical error - minimumA is bigger than desiredA");
                 panic_with_error!(env, ContractError::GetDepositAmountsMinABiggerThenDesiredA);
             }
         }
         if let Some(min_b) = min_b {
             if min_b > desired_b {
-                panic_with_error!(env, ContractError::GetDepositAmountsMinABiggerThenDesiredA);
+                log!(&env, "Pool: GetDepositAmounts: Critical error - minimumB is bigger than desiredB");
+                panic_with_error!(env, ContractError::GetDepositAmountsMinBBiggerThenDesiredB);
```
Comment (ueco-jb) [inline contracts/pool/src/storage.rs:263]: Good find. :+1:

**Review by ueco-jb — APPROVED** [review]
LGTM

**Pull request merged**